lib/server/swarm.js

```diff
@@ -7,12 +7,27 @@ var randomIterate = require('random-iterate')
 // Regard this as the default implementation of an interface that you
 // need to support when overriding Server.createSwarm() and Server.getSwarm()
 function Swarm (infoHash, server) {
+  this.complete = 0
+  this.incomplete = 0
   this.peers = new LRU({
     max: server.peersCacheLength || 1000,
-    maxAge: server.peersCacheTtl || 900000 // 900 000ms = 15 minutes
+    maxAge: server.peersCacheTtl || 20 * 60 * 1000 // 20 minutes
+  })
+
+  // When a websocket peer is evicted from the LRU cache, close the websocket
+  // after a short timeout period. We wait 1s so the server has a chance to send
+  // a response to 'stopped' events, which remove the peer and cause an eviction.
+  this.peers.on('evict', function (data) {
+    var peer = data.value
+    if (peer.socket) {
+      setTimeout(function () {
+        try {
+          peer.socket.close()
```

Comment:
Once the socket is closed by the server, there is no way to send messages to the client. So I think it's fine to set `peer.socket` to null immediately after we close the socket.

The timeout is only needed for the case where the client sends a 'stopped' message. When the server gets the message it calls `lru.remove()` which emits 'evict' immediately. We need the timeout so that the server can send a response with `socket.send()`.